app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt

```diff
@@ -230,6 +231,15 @@ class HistoryFragment : LibraryPageFragment<History>(), UserInteractionHandler,
         }
     }
 
+    override fun onPrepareMenu(menu: Menu) {
+        super.onPrepareMenu(menu)
+
+        if (historyStore.state.isEmpty) {
+            menu.findItem(R.id.history_search)?.isEnabled = false
+            menu.findItem(R.id.history_delete)?.isEnabled = false
+        }
```

Comment:
I was able to replicate the desired behavior here by moving this into `onCreateMenu`, and we could avoid this additional override.  This also allowed me to switch the `isEmpty` argument to `HistoryState` back to `false`. I assume that the dispatch and call to `invalidateMenu`() in `onEmptyStateChanged` is what is causing the correct behavior